### PR TITLE
fix nullpointer on recreate activity

### PR DESCRIPTION
**Pull request by gonzalonm**

### CODE REVIEW

**Comment by gonzalonm** [comment]
The approach was taken from here: https://goo.gl/FC6mWF

**Comment by gonzalonm** [comment]
Related bug:  https://bugzilla.xamarin.com/show_bug.cgi?id=42899

**Comment by pjcollins** [comment]
This fix works for me, though I'm wondering if there may be a better approach. In our style files, we’re defining themes with the same name as themes provided in `Android.Resource.Style` (e.g. `ThemeHoloDialog`). I believe this conflict could also cause Bug 42899, and is probably worth investigating?

**Comment by pjcollins** [comment]
LGTM 👍

File: ApiDemo/App/ActivityRecreate.cs [hunk 1]
```diff
@@ -35,23 +35,28 @@ protected override void OnCreate (Bundle bundle)
...
-				}
+		        switch (current_theme)
+		        {
+		            case Resource.Style.ThemeCurrent:
+		                current_theme = Resource.Style.ThemeCurrentDialog;
+		                break;
+		            case Resource.Style.ThemeCurrentDialog:
+		                current_theme = Resource.Style.ThemeCurrentDialogWhenLarge;
+		                break;
+                    case Resource.Style.ThemeCurrentDialogWhenLarge:
```
Comment (olegoid) [inline ApiDemo/App/ActivityRecreate.cs]: Something is wrong with tabulation here.

File: ApiDemo/App/ActivityRecreate.cs [hunk 2]
```diff
@@ -35,23 +35,28 @@ protected override void OnCreate (Bundle bundle)
...
+                        current_theme = Resource.Style.Theme;
+                        break;
+                    default:
+		                current_theme = Resource.Style.ThemeCurrent;
+		                break;
+		        }
 
-				SetTheme (current_theme);
-			}
+		        SetTheme(current_theme);
```
Comment (olegoid) [inline ApiDemo/App/ActivityRecreate.cs]: Something is wrong with tabulation here.

File: ApiDemo/App/ActivityRecreate.cs [hunk 3]
```diff
@@ -60,7 +65,7 @@ protected override void OnCreate (Bundle bundle)
 			button.Click += delegate { Recreate (); };
 		}
 
-		protected override void OnSaveInstanceState (Bundle outState)
```
Comment (olegoid) [inline ApiDemo/App/ActivityRecreate.cs]: Something is wrong with tabulation here.

**Comment by olegoid** [comment]
LGTM

**Pull request merged**